python/src/scipp/plotting/resampling_model.py

```diff
@@ -147,16 +165,18 @@ def _resample(self, array):
             # Must specify bounds for final dim despite handling by `histogram`
             # below: If coord is ragged binning would throw otherwise.
             bounds = sc.concatenate(edges[dim, 0], edges[dim, -1], dim)
-            binned = sc.bin_with_coords(array.data, array.meta,
-                                        self.edges[:-1] + [bounds], [])
+            binned = sc.bin(array=array, edges=self.edges[:-1] + [bounds])
             a = sc.histogram(binned, edges)
         else:
-            a = sc.bin_with_coords(array.data, array.meta, self.edges,
-                                   []).bins.sum()
+            a = sc.bin(array=array, edges=self.edges).bins.sum()
         for name, mask in array.masks.items():
             a.masks[name] = self._rebin(mask, array.meta)
         return a
 
+    def update_array(self, *args, **kwargs):
+        super().update_array(*args, **kwargs)
+        self._recenter_coords()
```

Comment:
Have you consider reverting the design, which might be slightly cleaner: Let the base-class `update_array()` call a method that can be overridden by child classes. Something like:

```python
# ResamplingModel
@array.setter
def array(self, array):
  self._array = array
  self._data_updated()

# child classes
def _data_updated(self):
   # whatever
```